x/superfluid/keeper/slash_test.go

```diff
@@ -198,3 +201,104 @@ func (suite *KeeperTestSuite) TestSlashLockupsForUnbondingDelegationSlash() {
+			suite.Require().NoError(err)
+			upperTickInfoPreSlash, err := suite.App.ConcentratedLiquidityKeeper.GetTickInfo(suite.Ctx, clPoolId, positionPreSlash.UpperTick)
+			suite.Require().NoError(err)
+			liquidityPreSlash := clPool.GetLiquidity()
+
+			// Calculate underlying assets from liquidity getting slashed
+			asset0PreSlash, asset1PreSlash, err := cl.CalculateUnderlyingAssetsFromPosition(suite.Ctx, positionPreSlash, clPool)
+			suite.Require().NoError(err)
+
+			slashAmt := positionPreSlash.Liquidity.Mul(tc.slashPercent)
+
+			// System under test
+			clPoolAddress, underlyingAssetsToSlash, err := suite.App.SuperfluidKeeper.PrepareConcentratedLockForSlash(suite.Ctx, lock, slashAmt)
+
+			lowerTickInfoPostSlash, getTickErr := suite.App.ConcentratedLiquidityKeeper.GetTickInfo(suite.Ctx, clPoolId, positionPreSlash.LowerTick)
+			suite.Require().NoError(getTickErr)
+			upperTickInfoPostSlash, getTickErr := suite.App.ConcentratedLiquidityKeeper.GetTickInfo(suite.Ctx, clPoolId, positionPreSlash.UpperTick)
+			suite.Require().NoError(getTickErr)
+
+			clPool, getPoolErr := suite.App.ConcentratedLiquidityKeeper.GetPoolFromPoolIdAndConvertToConcentrated(suite.Ctx, clPoolId)
+			suite.Require().NoError(getPoolErr)
+			liquidityPostSlash := clPool.GetLiquidity()
+			if tc.expectedErr {
+				suite.Require().Error(err)
+			} else {
+				suite.Require().NoError(err)
+				suite.Require().Equal(clPool.GetAddress(), clPoolAddress)
+
+				suite.Require().Equal(lowerTickInfoPreSlash.LiquidityGross.Sub(slashAmt).String(), lowerTickInfoPostSlash.LiquidityGross.String())
+				suite.Require().Equal(upperTickInfoPreSlash.LiquidityGross.Sub(slashAmt).String(), upperTickInfoPostSlash.LiquidityGross.String())
+
+				suite.Require().Equal(lowerTickInfoPreSlash.LiquidityNet.Sub(slashAmt).String(), lowerTickInfoPostSlash.LiquidityNet.String())
+				suite.Require().Equal(upperTickInfoPreSlash.LiquidityNet.Add(slashAmt).String(), upperTickInfoPostSlash.LiquidityNet.String())
+
+				suite.Require().Equal(liquidityPreSlash.Sub(slashAmt).String(), liquidityPostSlash.String())
+
+				positionPostSlash, err := suite.App.ConcentratedLiquidityKeeper.GetPosition(suite.Ctx, positionId)
+				suite.Require().NoError(err)
+				suite.Require().Equal(positionPreSlash.Liquidity.Sub(slashAmt).String(), positionPostSlash.Liquidity.String())
+
+				asset0PostSlash, asset1PostSlash, err := cl.CalculateUnderlyingAssetsFromPosition(suite.Ctx, positionPostSlash, clPool)
+				suite.Require().NoError(err)
+
+				errTolerance := osmomath.ErrTolerance{
+					AdditiveTolerance: sdk.NewDec(1),
+					RoundingDir:       osmomath.RoundDown,
+				}
+
+				suite.Require().Equal(0, errTolerance.Compare(asset0PreSlash.Sub(asset0PostSlash).Amount, underlyingAssetsToSlash[0].Amount))
+				suite.Require().Equal(0, errTolerance.Compare(asset1PreSlash.Sub(asset1PostSlash).Amount, underlyingAssetsToSlash[1].Amount))
```

Comment:
Please add fee and incentive accumulator update tests

Reply:
I am going to create an issue for this since I have some questions on what specifically we want to test and don't want it to block this PR

Reply:
Added issue https://github.com/osmosis-labs/osmosis/issues/4948

We can sync about this on next CL call